deepmd/train/trainer_mt.py

```diff
@@ -0,0 +1,713 @@
+#!/usr/bin/env python3
+import logging
+import os
+import time
+import shutil
+import google.protobuf.message
+import numpy as np
+from deepmd.env import tf
+from deepmd.env import default_tf_session_config
+from deepmd.env import GLOBAL_TF_FLOAT_PRECISION
+from deepmd.env import GLOBAL_ENER_FLOAT_PRECISION
+from deepmd.fit import EnerFitting, WFCFitting, PolarFittingLocFrame, PolarFittingSeA, GlobalPolarFittingSeA, DipoleFittingSeA
+from deepmd.descriptor import DescrptLocFrame
+from deepmd.descriptor import DescrptSeA
+from deepmd.descriptor import DescrptSeT
+from deepmd.descriptor import DescrptSeAEbd
+from deepmd.descriptor import DescrptSeAEf
+from deepmd.descriptor import DescrptSeR
+from deepmd.descriptor import DescrptSeAR
+from deepmd.descriptor import DescrptHybrid
+from deepmd.model import EnerModel, WFCModel, DipoleModel, PolarModel, GlobalPolarModel
+from deepmd.loss import EnerStdLoss, EnerDipoleLoss, TensorLoss
+from deepmd.utils.errors import GraphTooLargeError
+from deepmd.utils.learning_rate import LearningRateExp
+from deepmd.utils.neighbor_stat import NeighborStat
+from deepmd.utils.sess import run_sess
+from deepmd.utils.type_embed import TypeEmbedNet
+
+from tensorflow.python.client import timeline
+from deepmd.env import op_module
+
+# load grad of force module
+import deepmd.op
+
+from deepmd.common import j_must_have, ClassArg
+
+log = logging.getLogger(__name__)
+
+
+def _is_subdir(path, directory):
+    path = os.path.realpath(path)
+    directory = os.path.realpath(directory)
+    if path == directory:
+        return False
+    relative = os.path.relpath(path, directory) + os.sep
+    return not relative.startswith(os.pardir + os.sep)
+
+def _generate_descrpt_from_param_dict(descrpt_param):
```

Comment:
We'd better import `_generate_descrpt_from_param_dict` from `trainer.py` instead of writing a duplicated one.